server/service/src/item/bundled_item/upsert.rs

```diff
@@ -68,12 +74,57 @@ fn validate(
     connection: &StorageConnection,
     input: &UpsertBundledItem,
 ) -> Result<(), UpsertBundledItemError> {
-    if !check_item_variant_exists(connection, &input.principal_item_variant_id)? {
-        return Err(UpsertBundledItemError::PrincipalItemDoesNotExist);
+    let principal_item_variant =
+        match check_item_variant_exists(connection, &input.principal_item_variant_id)? {
+            Some(principal_item_variant) => principal_item_variant,
+            None => return Err(UpsertBundledItemError::PrincipalItemDoesNotExist),
+        };
+
+    let bundled_item_variant =
+        match check_item_variant_exists(connection, &input.bundled_item_variant_id)? {
+            Some(bundled_item_variant) => bundled_item_variant,
+            None => return Err(UpsertBundledItemError::BundledItemDoesNotExist),
+        };
+
+    if input.principal_item_variant_id == input.bundled_item_variant_id {
+        return Err(UpsertBundledItemError::CanNotBundleItemWithItself);
+    }
+
+    // Check that item_ids are not the same
+    // Technically this has a problem if the item is merged but should be very rare...
+    if principal_item_variant.item_link_id == bundled_item_variant.item_link_id {
```

Comment:
I ended up joining to item in #5324, if we merge those two, and use variant_repo rather than variant_row_repo in the `check_item_variant` call... can fix here 😅 

Aware/agree the risk is probably low here for now, but I'm a bit worried about the proliferation of item merge TODOs throughout the codebase, would love the get the right patterns in for us to start duplicating...

Reply:
> TODOs throughout the codebase, would love the get the right patterns in for us to start duplicating...

I have asked for this to be documented! @andreievg @roxy-dao any progress?

Reply:
Nope... sorry ;-;